### PR TITLE
Create tiny-jar for std-lib

### LINKED ISSUE #218: Pack std-lib as a separate jar for tiny jars version of Morganey
Derived from the discussion on the stream.

Just keep the std-lib in a separate jar `morganey-stdlib.jar`

### PR DESCRIPTION
**Pull request by keddelzz**
closes #218

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -32,12 +32,39 @@ lazy val dependencySettings =
...
+  stdLibJarFile
+}
+
 lazy val build = TaskKey[Unit]("build", "Builds morganey and copies all generated files into the target folder.")
 
 build := {
   val (_, coreFile) = packagedArtifact.in(Compile, packageBin).value
   val (_, macroFile) = packagedArtifact.in(macros).in(Compile, packageBin).value
   val (_, kernelFile) = packagedArtifact.in(kernel).in(Compile, packageBin).value
+  val _ = packStdLib.value
```
Comment (keddelzz) [inline build.sbt:67]: The assignment in this statement can safely be replaced by the method call (without assignment).

``` scala
packStdLib.value
```

Comment (ForNeVeR) [reply build.sbt:67]: I'm not sure, but could you maybe try to move this statement to `dependsOn` clause? Like `dependsOn(packStdLib, macros % dependencySettings, ...)`?

**Review by rexim — APPROVED** [review]
LGTM :+1:

**Comment by rexim** [comment]
Waiting on the @ForNeVeR's resolution

**Review by ForNeVeR — CHANGES_REQUESTED** [review]
I'd like @keddelzz to check my suggestion. If it won't work — then LGTM.

File: build.sbt (hunk 1 above)
Comment (keddelzz) [reply build.sbt:67]: This is not possible. `packStdLib` is of type `TaskKey[File]` and `dependsOn` requires a value of type `Project`. However dependencies on tasks can be expressed differently:

For example:

``` scala
myTask <<= (aTask, bTask) map { (a: A, b: B) =>
  ... do something with a, b and generate a result ...
}
```

Example taken from SBT-documentation [here](http://www.scala-sbt.org/0.12.1/docs/Detailed-Topics/TaskInputs.html).

Should I rewrite both tasks `build` and `packStdLib` in terms of this way to specify dependencies?

Comment (ForNeVeR) [reply build.sbt:67]: Okay then, I approve the current implementation. Forgive the old man for confusing you.

**Review by ForNeVeR — APPROVED** [review]

**Comment by rexim** [comment]
@keddelzz thanks for the changes!
@ForNeVeR thanks for the review!

**Pull request merged**